.template/addons/docker/Dockerfile.tt

```diff
@@ -1,14 +1,18 @@
 FROM ruby:<%= RUBY_VERSION %>-slim
 
 ARG RUBY_ENV=development
+<%- if WEB_VARIANT -%>
 ARG NODE_ENV=development
+<%- end -%>
```

Comment:
Why not using `insert_into_file Dockerfile` similarly as done for the Gemfile?

Reply:
For this one, I'm trying to make an addon for the docker as it is different between web and api. I am thinking about many ways that we could do for this one. 

1. Duplicate the files onto web and api, the con is that we need to maintain many files.
2. Use `insert_into_file`, but this one might be a lot to find where to insert, it is very scattered on the docker file. Going this way also make it hard to move docker out to addon. Apply it on the base would be the easiest as we can place the modifying thing on the variant folder.
3. Move docker to addon and use condition to modifying the file. (The current one) - This still feels weird as we usually do insert on the variant.

I think I can try to work on the insert thing to see how it looks like. 🤔 And we can discuss how should we do it.